en/advanced-topics/memory-leak-detector.md

```diff
@@ -0,0 +1,53 @@
+# Native engine memory leak detection system
+
+The native engine is developed using the C++ language, and memory leaks are inevitable. In order to facilitate game & engine developers to quickly find memory leaks, this version provides a **memory leak detection system**
+
+## 1, Compared with other memory leak detection tools, what are the advantages of the built-in memory leak detection tool in cocos:
+- Cross-platform: support windows/android/mac/ios platform
+- Ease of use: No need to download additional tools and perform complex configurations; support output of stack information at memory leaks, which is convenient for quickly locating leaks
+- Consistency: The usage process of each platform is almost the same: start the game from the ide -> run for a period of time -> close the game -> view the ide output log
+- Real-time: Although the frame rate of the game in the profiling mode has dropped, it still maintains the real-time running frame rate
+- Accuracy: theoretically zero false positives
+
+## 2，Steps for usage
+- Modify the macro USE_MEMORY_LEAK_DETECTOR in cocos/base/Config.h to 1, the default value of 0 means that the memory leak detection system is turned off, and there will be zero loss of engine performance after it is turned off
+    ```ts
+    #ifndef USE_MEMORY_LEAK_DETECTOR
+        #define USE_MEMORY_LEAK_DETECTOR 1
+    #endif
+    ```
+- Two additional things need to be done on the android platform:
+    - Add a line as follows in the project directory/native/engine/android/CMakeLists.txt:
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -finstrument-functions")
+        ```ts
+        set(PROJ_SOURCES
+            ${CMAKE_CURRENT_LIST_DIR}/../common/Classes/Game.h
+            ${CMAKE_CURRENT_LIST_DIR}/../common/Classes/Game.cpp
+            ${CMAKE_CURRENT_LIST_DIR}/jni/main.cpp
+        )
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -finstrument-functions")
+        ```
+    - If the Job System is TBB when building the android project, please change to None or TaskFlow (default), because TBB will conflict with the current memory capture implementation mechanism on the android platform
```

Comment:
```suggestion
- If the Job System is TBB when building the android project, please change to None or TaskFlow (default), because TBB will conflict with the current memory capture implementation mechanism on the android platform:
```

Reply:
这里要缩进